Python/suggestions.c

```diff
@@ -171,6 +171,17 @@ offer_suggestions_for_name_error(PyNameErrorObject *exc) {
     }
     suggestions = calculate_suggestions(dir, name);
     Py_DECREF(dir);
+    if (suggestions != NULL) {
+        return suggestions;
+    }
+
+    dir = PySequence_List(frame->f_builtins);
+    if (dir == NULL) {
+        return NULL;
+    }
+    assert(PyList_GET_SIZE(dir) <= MAX_CANDIDATE_ITEMS);
+    suggestions = calculate_suggestions(dir, name);
```

Comment:
People can mutate builtins dictionary, so this assertion will crash interpreters on debug mode if some stuff arbitrarily gets inserted into the builtins. I think it would be better off if you would just guard it.
```suggestion
    if (PyList_GET_SIZE(dir) <= MAX_CANDIDATE_ITEMS) {
        suggestions = calculate_suggestions(dir, name);
    }
```

Reply:
Here is a simple crasher;
```py
import builtins

def foo():
    for number in range(100):
        setattr(builtins, 'a' * number, number)
    
    aaaaac

foo()
```
```
 $ ./python t.py
Traceback (most recent call last):
  File "/home/isidentical/cpython/cpython/t.py", line 9, in <module>
    foo()
  File "/home/isidentical/cpython/cpython/t.py", line 7, in foo
    aaaaac
python: Python/suggestions.c:182: offer_suggestions_for_name_error: Assertion `PyList_GET_SIZE(dir) <= MAX_CANDIDATE_ITEMS' failed.
[1]    49965 abort (core dumped)  ./python t.py
```

Reply:
Sup, that assert was not supposed to be there, it was just me checking things :sweat_smile: 

I have added a test and remove that assert :)

Reply:
Haha, cool!